src/snowflake/snowpark/_internal/type_utils.py

```diff
@@ -608,6 +644,22 @@ def python_type_str_to_object(
         return eval(tp_str)
 
 
+def sql_server_type_to_snow_type(
+    tp: Tuple[Tuple[str, str, int, int, int, int, str]],
+) -> DataType:
+    snow_type = SQL_SERVER_TYPE_TO_SNOW_TYPE.get(tp[1].lower(), None)
+    if snow_type is None:
+        raise ValueError(f"sql server type not supported: {tp}")
```

Comment:
nit: NotImplementedError might be better
```suggestion
    if snow_type is None:
        raise NotImplementedError(f"sql server type not supported: {tp}")
```